pkg/policy/snapshots.go

```diff
@@ -0,0 +1,228 @@
+package policy
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"regexp"
+	"strings"
+
+	"github.com/jeremywohl/flatten"
+)
+
+func (ce *Executor) StoreSnapshot(ctx context.Context, path string, tables []string) error {
+	if len(tables) == 0 {
+		return errors.New("no tables to snapshot")
+	}
+
+	for _, table := range tables {
+		ef, err := os.OpenFile(fmt.Sprintf("%s/table_%s.csv", path, table), os.O_CREATE|os.O_WRONLY, 0777)
+		if err != nil {
+			ce.log.Error("error opening file.", "error", err)
+			return err
+		}
+		defer ef.Close()
+		query := fmt.Sprintf("COPY (select * from %s) TO STDOUT DELIMITER '|' CSV HEADER", table)
+		ce.log.Debug("exporting", "table", table, "query", query)
+
+		err = ce.storeOutput(ctx, ef, query)
+		if err != nil {
+			ce.log.Error("error exporting file.", "error", err)
+			return fmt.Errorf("error exporting file: %+v", err)
+		}
+
+	}
+
+	return nil
+}
+
+func (ce *Executor) RestoreSnapshot(ctx context.Context, source string) error {
+
+	ef, err := os.OpenFile(source, os.O_RDONLY, 0777)
+	if err != nil {
+		ce.log.Error("error opening file.", "error", err)
+		return err
+	}
+	defer ef.Close()
+	fileName := path.Base(source)
+	if !strings.HasPrefix(fileName, "table_") || !strings.HasSuffix(fileName, ".csv") {
+		ce.log.Error("truncating", "source", source, "file", fileName)
+		return errors.New("invalid filename")
+	}
+	source_name := strings.TrimPrefix(strings.TrimSuffix(fileName, ".csv"), "table_")
+	truncQuery := fmt.Sprintf("TRUNCATE %s CASCADE", source_name)
+	ce.log.Debug("truncating", "table", source_name, "query", truncQuery)
+	err = ce.conn.Exec(ctx, truncQuery)
+	if err != nil {
+		ce.log.Error("error importing data from file.", "error", err, "source", source)
+		return fmt.Errorf("error importing file: %+v", err)
```

Comment:
```suggestion
		return fmt.Errorf("error importing file %q: %w", fileName, err)
```
could maybe remove the filename from the print, not sure (as it will show up in sentry)
plus remove the log :)